mcs/class/Mono.Posix/Mono.Unix.Native/Stdlib.cs

```diff
@@ -426,11 +426,21 @@ static Stdlib ()
 
 		public static Errno GetLastError ()
 		{
-			int errno = Marshal.GetLastWin32Error ();
-			return NativeConvert.ToErrno (errno);
+			if (Environment.OSVersion.Platform != PlatformID.Unix) {
```

Comment:
`Environment.OSVersion` is lazy loaded.  So if it hasn't been loaded the first time `GetLastError` is invoked, it ends up calling `uname`, which will clear the errno. Thus errno is no longer set in order to retrieve the last error.

@mjrousos ran into this issue using the latest code.